### PR TITLE
Update npm dev dependencies

### PR DESCRIPTION
**Pull request by lukasbestle**
- Temporarily pinned the code dependency versions to their current versions so that they can be safely updated one by one, see https://github.com/getkirby/kirby/issues/2930.
- Updated all npm dev dependencies (except Cypress) to their latest versions
  - `node-sass` is deprecated and was replaced with `dart-sass` (the `sass` npm package)
  - `babel-core` was renamed to `@babel/core`
  - The upgrades mean that our setup is now compatible with Node v15
- The Vue CLI ESLint setup changed quite a bit. Might be that our old setup was broken or just that the new setup has new rules, but there were suddenly hundreds of errors and warnings. I have run the auto-fixer and also made manual changes to fix the fatal errors. 35 warnings still remain in `vue-cli-service lint` that we should review one by one.

**TODO:**

- [ ] Rebase against `features` when all other PRs are merged
- [ ] Run `npm update`
- [ ] Run `node_modules/.bin/vue-cli-service lint --fix`

### CODE REVIEW

**Comment by bastianallgeier** [comment]
I wonder if we should take the eslint fixes out of this and move them to the very end before we release the first rc for 3.5. Otherwise those linting changes might make it super complicated to merge features because of dozens of conflicts.

**Comment by lukasbestle** [comment]
That's true. In this case we should also wait with the dependency updates as the errors fixed in this PR are fatal errors. Vue CLI won't build without fixing them. So let's wait until all other PRs are merged and then I'll rebase this.

**Comment by lukasbestle** [comment]
Meanwhile we could already start reviewing the other package updates (#2930) and add the changes to this PR. Then we won't have to do that so close to the release. @bastianallgeier @distantnative

**Comment by lukasbestle** [comment]
I have removed the automatic ESLint fixes and the lockfile changes from this PR as these will definitely collect merge conflicts over time. I added a TODO list to the first post of this PR.

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**
